ddtrace/internal/_rand.pyx

```diff
@@ -0,0 +1,56 @@
+
+To put this into perspective: we cap the max number of traces at 1k/s let's be
+conservative and say each trace contains 100 spans.
+
+That's 100k spans/second which would be 100k + 1 calls to this fn per second.
+
+That's 10,000,000 seconds until we hit the period. That's 115 days of
+100k spans/second (with no application restart) until the period is reached.
+
+
+rand64bits() is thread-safe as it is written in C and is interfaced with via
+a single Python step. This is the same mechanism in which CPython achieves
+thread-safety:
+https://github.com/python/cpython/blob/8d21aa21f2cbc6d50aab3f420bb23be1d081dac4/Lib/random.py#L37-L38
+
+
+Python 2.7:
+Name (time in ns)                         Min                   Max                  Mean              StdDev                Median                 IQR            Outliers  OPS (Kops/s)            Rounds  Iterations
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+rand64bits                           144.1789 (1.01)       221.2596 (1.0)        155.5800 (1.0)       15.4198 (1.0)        151.3100 (1.00)       7.6687 (1.0)           4;6    6,427.5628 (1.0)          61      100000
+random.SystemRandom().getrandbits  1,626.8015 (11.37)    2,178.9074 (9.85)     1,766.1762 (11.35)    133.8990 (8.68)     1,714.4561 (11.35)    113.9164 (14.85)        11;8      566.1949 (0.09)         60       10000
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+
+Python 3.7:
+Name (time in ns)                       Min                 Max                Mean             StdDev              Median                IQR            Outliers  OPS (Mops/s)            Rounds  Iterations
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+rand64bits                         167.5956 (1.0)      211.3155 (1.0)      190.2803 (1.0)       9.5815 (1.0)      187.7187 (1.0)      11.4513 (1.0)          15;1        5.2554 (1.0)          52      100000
+random.randbits                    222.7103 (1.33)     367.4459 (1.74)     250.2699 (1.32)     26.5930 (2.78)     242.1607 (1.29)     26.4550 (2.31)          6;1        3.9957 (0.76)         36      100000
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+"""
+from libc.stdint cimport uint64_t
+
+from ddtrace import compat
+
+
+cdef uint64_t x = compat.getrandbits(64) ^ 4101842887655102017
+
+
+cpdef public uint64_t rand64bits():
```

Comment:
```suggestion
def public uint64_t rand64bits():
```
This is never used from C, so you can just use `def` here.

Reply:
:+1: out of curiosity (and for the record) I benchmarked `def` vs `cdef` vs `cpdef` to see if there would be any difference. There wasn't:

```
--------------------------------------------------------------------------------------- benchmark 'span-id': 3 tests ---------------------------------------------------------------------------------------
Name (time in ns)                      Min                 Max                Mean             StdDev              Median                IQR            Outliers  OPS (Mops/s)            Rounds  Iterations
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_span_id_rand64bits_cpdef     185.6399 (1.0)      411.4294 (1.15)     229.6607 (1.03)     61.4212 (2.04)     203.8789 (1.0)      31.4218 (1.22)         8;10        4.3543 (0.98)         51      100000
test_span_id_rand64bits_cdef      188.3510 (1.01)     584.9214 (1.63)     224.3294 (1.00)     54.0005 (1.80)     204.3416 (1.00)     26.9241 (1.04)        19;26        4.4577 (1.00)        181       25824
test_span_id_rand64bits_def       196.7382 (1.06)     359.1490 (1.0)      224.0553 (1.0)      30.0369 (1.0)      215.6258 (1.06)     25.8040 (1.0)           7;3        4.4632 (1.0)          48      100000
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```